docs/zh_cn/01-how-to-build/snpe.md

```diff
@@ -0,0 +1,193 @@
+# 支持 SNPE
+
+mmdeploy 集成 snpe 的方式简单且有效： Client/Server 模式。
+
+这种模式
+
+1. 能剥离`模型转换`和`推理`环境：
+
+- 推理无关事项在算力更高的设备上完成；
+- 对于推理计算，能拿到 gpu/npu 真实运行结果，而非 cpu 模拟数值。
+
+2. 能覆盖成本敏感的设备。armv7/risc-v/mips 芯片满足产品需求，但往往对 Python 支持有限；
+
+3. 能简化 mmdeploy 安装步骤。如果只想转 snpe 模型测试精度，不需要编译 .whl 包。
+
+## 一、运行推理服务
+
+下载预编译 snpe 推理服务包， `adb push` 到手机、执行。
+注意**手机要有 qcom 芯片**。
+
+```bash
+$ wget https://media.githubusercontent.com/media/tpoisonooo/mmdeploy_snpe_testdata/main/snpe-inference-server-1.59.tar.gz
+...
+$ sudo apt install adb
+$ adb push snpe-inference-server-1.59.tar.gz  /data/local/tmp/
+
+# 解压运行
+$ adb shell
+venus:/ $ cd /data/local/tmp
+130|venus:/data/local/tmp $ tar xvf snpe-inference-server-1.59.tar.gz
+...
+130|venus:/data/local/tmp $ source export1.59.sh
+130|venus:/data/local/tmp $ ./inference_server
+...
+  Server listening on [::]:50052
+```
+
+此时推理服务应打印设备所有 ipv6 和 ipv4 地址，并监听端口。
+
+tips:
+
+- 如果 `adb devices` 找不到设备，可能因为：
+  - 有些廉价线只能充电、不能传输数据
+  - 或者没有打开手机的“开发者模式”
+- 如果需要自己编译，可参照 [NDK 交叉编译 snpe 推理服务](../appendix/cross_build_snpe_service.md)
+
+## 二、安装 mmdeploy
+
+1. 环境要求
+
+| 事项    | 版本               | 备注          |
+| ------- | ------------------ | ------------- |
+| host OS | ubuntu18.04 x86_64 | snpe 指定版本 |
+| Python  | **3.6.0**          | snpe 指定版本 |
+
+2. 安装
+
+[官网下载 snpe-1.59](https://developer.qualcomm.com/qfile/69652/snpe-1.59.0.zip)，解压设置环境变量
+
+```bash
+$ unzip snpe-1.59.0.zip
+$ export SNPE_ROOT=${PWD}/snpe-1.59.0.3230
+$ cd /path/to/mmdeploy
+$ export PYTHONPATH=${PWD}/service/snpe/client:${SNPE_ROOT}/lib/python:${PYTHONPATH}
+$ export LD_LIBRARY_PATH=${SNPE_ROOT}/lib/x86_64-linux-clang:${LD_LIBRARY_PATH}
+$ python3 -m pip install -e .
+```
+
+tips:
+
+- 如果网络不好，[这个 .tar.gz](https://github.com/tpoisonooo/mmdeploy_snpe_testdata/blob/main/snpe-1.59.tar.gz) 仅减小官方包体积，没有修改原始内容。
+
+## 三、测试模型
+
+以 Resnet-18 为例。先参照[文档安装 mmcls](https://github.com/open-mmlab/mmclassification)，然后使用 `tools/deploy.py` 转换模型。
+
+```bash
+$ export MODEL_CONFIG=/path/to/mmclassification/configs/resnet/resnet18_8xb16_cifar10.py
+$ export MODEL_PATH=https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_b16x8_cifar10_20210528-bd6371c8.pth
+
+# 模型转换
+$ cd /path/to/mmdeploy
+$ python3 tools/deploy.py  configs/mmcls/classification_snpe_dynamic.py $MODEL_CONFIG  $MODEL_PATH   /path/to/test.png   --work-dir resnet18   --device cpu  --uri 10.0.0.1\:50051  --dump-info
```

Comment:
change `configs/mmcls/classification_snpe_dynamic.py` to static one.

Reply:
Done, updated here https://github.com/open-mmlab/mmdeploy/pull/789/commits/27b1b6df